src/test/ui/static/static-reference-to-fn-2.stderr

```diff
@@ -40,6 +40,14 @@ LL | |         statefn: &id(state1 as StateMachineFunc)
    | |                   ------------------------------ temporary value created here
 LL | |     }
    | |_____^ returns a value referencing data owned by the current function
+   |
+help: use `.collect()` to allocate the iterator
+   |
+LL |     StateMachineIter {
+LL |
+LL |         statefn: &id(state1 as StateMachineFunc)
+LL |     }.collect::<Vec<_>>()
+   |
```

Comment:
Because I use a different span now, I get this message misleading message.

Reply:
I'm using `span_suggestion_hidden` to hide the code.

Reply:
Should this diagnostic be triggering for this case at all? I can understand how it does given the current implementation, but it doesn't seem like I'd want it to as the user. I'm not sure if `span_suggestion_hidden` is intended to be used for cases like this.